prime-angular-frontend/src/app/modules/adjudication/pages/health-authorities/health-auth-care-types-page/health-auth-care-types-page.component.ts

```diff
@@ -65,7 +65,11 @@ export class HealthAuthCareTypesPageComponent implements OnInit {
   }
 
   public removeCareType(index: number) {
-    this.careTypes.removeAt(index);
+    const careType = this.careTypes.value[index].careType;
```

Comment:
If we are doing a check to see in the front end if you can remove a vendor/caretype you will want to give the user feedback as to why their action didn't work.

We have a couple routes we could take.
One option you could just not let the user have the option to do the action so instead of checking in this call you could check once on load of the page for each care type and make the "X" disabled (maybe add a tooltip as to why when they hover over).
The other option is the one which path you are on, you just need to relay the message to the user that the vendor/caretype is in use either through a toast or other device.

Reply:
@hogmann on that note, should we disable the dropdown too so that an admin cannot make a change if a vendor/care-setting is in use?

Reply:
Yeah understandable, the way the ticket was worded it made sense it led to the solution in the PR. Just wanted to put this comment as an alternative to that style of approach. Since the majority of times we want to guide the user to perform the actions they are allowed to rather than slap their wrist when they do an action they are not allowed to do.

Reply:
Feel free to close this conversation at any time